src/openms/source/FORMAT/DATAACCESS/MSDataCachedConsumer.cpp

```diff
@@ -78,7 +78,16 @@ namespace OpenMS
     }
     writeSpectrum_(s, ofs_);
     spectra_written_++;
-    if (clearData_) { s.clear(false); }
+
+    // Clear all spectral data including all float/int data arrays
+    if (clearData_)
+    {
+      s.clear(false);
+      MSSpectrum::FloatDataArrays fempty;
+      s.setFloatDataArrays(fempty);
```

Comment:
would s.setFloatDataArrays({}); also work?

Reply:
yes, it does!